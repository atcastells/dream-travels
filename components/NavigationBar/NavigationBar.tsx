import Image from "next/image";
import { Button } from "../Button/Button";

interface NavigationBarProps {
  onCreateTrip: () => void;
}

export const NavigationBar = ({ onCreateTrip }: NavigationBarProps) => {
  return (
    <div className="rounded-3xl p-4 mx-4 my-8 bg-black h-20 text-white flex justify-between">
      <Image src="/favicon.png" alt="logo" width={48} height={48} />
      <Button className="text-black bg-white" onClick={onCreateTrip}>
        <span className="m-4">Create new trip</span>
      </Button>
    </div>
  );
};
