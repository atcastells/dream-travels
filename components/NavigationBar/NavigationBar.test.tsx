import { expect, jest, test } from "@jest/globals";

import { render, screen } from "@testing-library/react";
import { NavigationBar } from "./NavigationBar";

test("NavigationBar renders", () => {
  const onCreateTripSpy = jest.fn();

  render(<NavigationBar onCreateTrip={onCreateTripSpy} />);
  expect(screen.getByText("Create new trip")).toBeTruthy();
  expect(screen.getByRole("button")).toBeTruthy();

  screen.getByRole("button").click();

  expect(onCreateTripSpy).toHaveBeenCalledTimes(1);

  //Snapshot testing

  expect(screen.getByRole("button")).toMatchSnapshot();
});
