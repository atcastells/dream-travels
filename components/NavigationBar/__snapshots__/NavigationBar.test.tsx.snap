// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders 1`] = `
<button
  class="text-black bg-white rounded-full px-2"
>
  <span
    class="m-4"
  >
    Create new trip
  </span>
</button>
`;
